### PR TITLE
Add AppVeyor for Windows CI

### LINKED ISSUE #625: Add windows CI using appveyor
We are using circle CI to run specs, circle CI do no have support for windows builds, having a windows CI will ensure windows compatibility of the application, 

Reference: https://www.appveyor.com/docs/lang/ruby/

### PR DESCRIPTION
**Pull request by fluctix**
Fixes #625 

#### Describe the changes you have made in this pr -
Added `appveyor.yml` for AppveyorCI

### CODE REVIEW

**Comment by fluctix** [comment]
@tachyons Coveralls Token should be added in GUI

File: config/database.example.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ default: &default
   encoding: unicode
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
   username: postgres
-  password: postgres
+  #password: postgres ##Uncomment This
```
Comment (ankitkataria) [inline config/database.example.yml]: Is this change required?

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+before_test:
+  - ruby -v
+  - gem -v
+  - bundle -v
+  - copy config\database.example.yml config\database.yml
+  - ruby bin\rails db:setup
+
+
+test_script:
+  - bundle exec coveralls push
```
Comment (ankitkataria) [inline appveyor.yml]: I've enabled the appveyor plugin for this repository, we can add the `bundle exec rspec` command here and test if the builds are running successfully.

File: config/database.example.yml (hunk 1 above)
Comment (fluctix) [reply config/database.example.yml]: I got password errors as Appveyor has Postgresql Password 'Password12!' , and as we copy the example database config it uses postgres as the password

File: appveyor.yml (hunk 2 above)
Comment (fluctix) [reply appveyor.yml]: `bundle exec coveralls push` also executes rspec

Comment (tachyons) [reply appveyor.yml]: Coveralls need to run after rspec command

**Comment by fluctix** [comment]
@ankitkataria is circleci disabled?

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+cache:
+  - vendor/bundle
+
+environment:
+  RAILS_ENV: test
+  PGHOST: 127.0.0.1
+  PGUSER: postgres
+  POSTGRES_DB: circuitverse_test
+  PGPASSWORD: Password12!
```
Comment (tachyons) [inline appveyor.yml]: I think you can update password here for consistency

Comment (fluctix) [reply appveyor.yml]: I don't think it will change the password for the user postgres

File: appveyor.yml (hunk 2 above)
Comment (fluctix) [reply appveyor.yml]: I believe it will ,their docs say,

> To do this with RSpec, just type bundle exec coveralls push in your project directory.
> 
> This will run RSpec and upload the coverage data to Coveralls as a one-off build

Comment (fluctix) [reply appveyor.yml]: Btw I have to say I tried running rspec directly on appveyor and it takes a lot of time(more than 10 mins)

File: appveyor.yml (hunk 3 above)
Comment (ankitkataria) [reply appveyor.yml]: We can create another database config file `database.appveyor.yml` and use that one instead of `database.example.yml`

**Comment by ankitkataria** [comment]
https://ci.appveyor.com/project/fluctix/circuitverse/builds/28999814

It seems you haven't tried coveralls in your cloned repo. I added it for our repo and the build seems to be failing https://ci.appveyor.com/project/ankitkataria/circuitverse/builds/28999816 Could you try setting up coveralls for your cloned repo and ensuring a successful build with the tests on this branch?

File: config/database.appveyor.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  username: postgres
+  password: Password12!
+  host: localhost
+
+
+development:
```
Comment (ankitkataria) [inline config/database.appveyor.yml]: I guess the `development` and `production`  sections will not be relevant for this database config.

**Comment by fluctix** [comment]
I can't figure out why the tests take so long
